src/esp/gfx/MaterialUtil.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "MaterialUtil.h"
+
+#include <Magnum/Trade/PbrMetallicRoughnessMaterialData.h>
+#include <Magnum/Trade/PhongMaterialData.h>
+
+namespace Mn = Magnum;
+
+namespace esp {
+namespace gfx {
+
+gfx::PhongMaterialData::uptr buildPhongFromPbrMetallicRoughness(
+    const Mn::Trade::PbrMetallicRoughnessMaterialData& material,
+    int textureBaseIndex,
+    const std::vector<std::shared_ptr<Magnum::GL::Texture2D>>& textures) {
+  // NOLINTNEXTLINE(google-build-using-namespace)
+  using namespace Mn::Math::Literals;
+
+  auto finalMaterial = gfx::PhongMaterialData::create_unique();
+
+  // If there's a roughness texture, we have no way to use it here. The safest
+  // fallback is to assume roughness == 1, thus producing no spec highlights.
+  const float roughness =
+      material.hasRoughnessTexture() ? 1 : material.roughness();
+
+  // If there's a metalness texture, we have no way to use it here. The safest
+  // fallback is to assume non-metal.
+  const float metalness =
+      material.hasMetalnessTexture() ? 0 : material.metalness();
+
+  // Heuristic to map roughness to spec power.
+  // Higher exponent makes the spec highlight larger (lower power)
+  // https://www.wolframalpha.com/input/?i=5+%2B+%281+-+x%29%5E1.5+*+75+for+x+from+0+to+1
+  // lower power for metal
+  const float maxShininess = Magnum::Math::lerp(250, 120, metalness);
```

Comment:
Wow! This can be rather useful. 
Where do you find these heuristics?
Do you have links?

Reply:
Nope, just brute-force tuned it myself using Blender as a reference. See my comparison images: https://github.com/facebookresearch/habitat-sim/pull/792#issuecomment-691774215

Reply:
Amazing!